lib/router/index.js

```diff
@@ -59,6 +61,22 @@ function throwIfNoRouter () {
 // Export the SingletonRouter and this is the public API.
 export default SingletonRouter
 
+export function withRouter (Comp) {
+  const compName = Comp.displayName || Comp.name || 'Component'
+
+  return class ComponentWithRouter extends Component {
+    static contextTypes = {
+      router: PropTypes.object
+    }
+
+    static displayName = `withRouter(${compName})`
+
+    render () {
+      return <Comp {...this.props} router={this.context.router} />
```

Comment:
Interesting, I hadn't seen that before. I don't understand how we would use createEagerFactory in this case, though. Wouldn't we use [createEagerElement](https://github.com/acdlite/recompose/blob/master/docs/API.md#createeagerelement) instead of JSX for the render return value?